Tools/F_scripts/write_probin.py

```diff
@@ -385,21 +364,111 @@ def write_probin(probin_template, param_A_files, param_B_files,
                             fout.write(", ")
 
+            elif keyword == "cxx_gets":
+                # this writes out the Fortran functions that can be
+                # called from C++ to get the value of the parameters
+
+                for p in params:
+                    if p.dtype == "character":
+                        # we don't support character
+                        continue
+                    if p.dtype == "logical":
+                        # F90 logicals are integers in C++
+                        fout.write("{}subroutine get_f90_{}({}_in) bind(C, name=\"get_f90_{}\")\n".format(
+                            indent, p.var, p.var, p.var))
+                        fout.write("{}   integer, intent(inout) :: {}_in\n".format(
+                            indent, p.var))
+                        fout.write("{}   {}_in = 0\n".format(indent, p.var))
+                        fout.write("{}   if ({}) then\n".format(indent, p.var))
+                        fout.write("{}      {}_in = 1\n".format(indent, p.var))
+                        fout.write("{}   endif\n".format(indent))
+                        fout.write("{}end subroutine get_f90_{}\n\n".format(
+                            indent, p.var))
+
+                    else:
+                        fout.write("{}subroutine get_f90_{}({}_in) bind(C, name=\"get_f90_{}\")\n".format(
+                            indent, p.var, p.var, p.var))
+                        fout.write("{}   {}, intent(inout) :: {}_in\n".format(
+                            indent, p.get_f90_decl(), p.var))
+                        fout.write("{}   {}_in = {}\n".format(
+                            indent, p.var, p.var))
+                        fout.write("{}end subroutine get_f90_{}\n\n".format(
+                            indent, p.var))
+
+
         else:
             fout.write(line)
 
     print(" ")
     fout.close()
 
+    # now handle the C++ -- we need to write a header and a .cpp file
+    # for the parameters + a _F.H file for the Fortran communication
 
-if __name__ == "__main__":
+    # first the _F.H file
+    ofile = "{}_parameters_F.H".format(cxx_prefix)
+    with open(ofile, "w") as fout:
+        fout.write(CXX_F_HEADER)
+
+        for p in params:
+            if p.dtype == "character":
+                # we don't support character
+                continue
+
+            fout.write("  void get_f90_{}({}* {});\n\n".format(
+                p.var, p.get_cxx_decl(), p.var))
+
+        fout.write(CXX_F_FOOTER)
+
+    # now the main C++ header with the global data
+    ofile = "{}_parameters.H".format(cxx_prefix)
+    with open(ofile, "w") as fout:
+        fout.write(CXX_HEADER)
+
+        fout.write("  void init_{}_parameters();\n\n".format(cxx_prefix))
+
+        for p in params:
+            if p.dtype == "character":
+                # we don't support character
+                continue
+
+            fout.write("  extern {} {};\n\n".format(p.get_cxx_decl(), p.var))
```

Comment:
We'll want to add a __managed__ qualifier so we can access these on the GPU, but we can do that in a later commit.

Reply:
Done.